docs/apm-mutating-webhook.asciidoc

```diff
@@ -0,0 +1,230 @@
+
+. <<apm-webhook-clone-repo>>
+. <<apm-webhook-configure-helm>>
+. <<apm-webhook-install-helm>>
+. <<apm-webhook-add-pod-annotation>>
+// Anything next? Restart the deployment maybe?
+
+[float]
+[[apm-webhook-clone-repo]]
+=== Clone the repository
+
+Clone the `elastic/apm-mutating-webhook` repository:
+
+[source,bash]
+----
+git clone git@github.com:elastic/apm-mutating-webhook.git
+cd apm-mutating-webhook
+----
+
+[float]
+[[apm-webhook-configure-helm]]
+=== Configure the webhook with a Helm values file
+
+The webhook is installed with a Helm chart.
+You can provide a custom webhook configuration using a Helm values file.
+Elastic provides a https://github.com/elastic/apm-mutating-webhook/blob/main/custom.yaml[`custom.yaml`] file as a starting point.
+
+Here's an example of a `custom.yaml` file that instruments a pod with the Elastic APM Java agent:
+
+[source,yaml]
+----
+apm:
+  secret_token: SuP3RT0K3N <1>
+  namespaces: <2>
+    - default
+    - my-name-space-01
+    - my-name-space-02
+webhookConfig:
+  agents:
+    java:
+      image: docker.elastic.co/observability/apm-agent-java:latest
+      artifact: "/usr/agent/elastic-apm-agent.jar"
+      environment:
+        JAVA_TOOL_OPTIONS: "-javaagent:/elastic/apm/agent/elastic-apm-agent.jar"
+        ELASTIC_APM_SERVER_URL: "https://apm-example.com:8200" <3>
+        ELASTIC_APM_ENVIRONMENT: "prod"
+        ELASTIC_APM_LOG_LEVEL: "info"
+----
+<1> The `secret_token` for your deployment. Use `api_key` if using an API key instead.
+<2> If you're using a secret token or API key to secure your deployment, you must list
+all of the namespaces where you want to auto-instrument pods. The secret token or API key
+will be stored as Kubernetes Secrets in each namespace.
+<3> Elastic APM agent environment variables—for example, the APM Server URL, which specifies the URL and port of your APM integration or server.
+
+NOTE:: The `artifact` and `JAVA_TOOL_OPTIONS` keys should not be edited.
+
+[float]
+[[apm-webhook-install-helm]]
+=== Install the webhook with Helm
+
+Install the webhook with Helm.
+Pass in your `custom.yaml` configuration file created in the previous step with the `--values` flag.
+
+[source,bash]
+----
+helm upgrade \
+  --install webhook apm-agent-auto-attach/ \
+  --namespace=elastic-apm \
+  --create-namespace \
+  --values custom.yaml
+----
+
+[float]
+[[apm-webhook-add-pod-annotation]]
+=== Add a pod template annotation to each pod you want to auto-instrument
+
+To auto-instrument a deployment, update its `spec.template.metadata.annotations` to include the
+`co.elastic.traces/agent` key. The webhook matches the value of this key to the `webhookConfig.agents`
+value defined in your Helm values file.
+
+For example, if your Webhook values file includes the following:
+
+[source,yaml]
+----
+...
+webhookConfig:
+  agents:
+    java:
+...
+----
+
+Then your `co.elastic.traces/agent` value should be `java`:
+
+[source,yaml]
```

Comment:
This seems like an awful lot of sample code to include in the documentation. There's only one line here that is relevant to the user. Can we get rid of the excess? Or does a full context example seem worthwhile?

Reply:
> Or does a full context example seem worthwhile?

Datapoint of one here, but I do think _some_ context example is important here, if only to re-enforce that the `spec.template.metadata.annotations` value you're adding is something that should be added to the YAML you're deploying to setup your pods and not a part of `custom.yaml`, `values.yaml`, or some other K8s configuration. 

I don't know if it fits in the elastic style guide, but I've always been a fan of commented ellipses as a way to simplify large context examples -- something like 

```
----
apiVersion: apps/v1
kind: Deployment
metadata:
  # ...
spec:
  replicas: 1
  template:
    metadata:
      # APM Mutating WebHook configuration
      annotations:
        co.elastic.traces/agent: java <1>
   # ...
----   
```

This makes it clear that you're talking about a `Deployment` configuration and that you're adding the `spec.template.metadata.annotations` section _without_ the reader needing to parse the rest of the configuration themselves.

Reply:
Yup, we do this elsewhere in the docs. I'll update to this.